std/algorithm/iteration.d

```diff
@@ -1034,6 +1049,96 @@ template each(alias pred = "a")
     assert(res == [9, 12, 15]);
 }
 
+// #15357: `each` should behave similar to foreach
+/// each should work with index
```

Comment:
Since it's a ddoc-ed example, "`each` works with iterable objects which provide an index variable, along with each element", or sth. like that would be more appropriate.